unumpy/_multimethods.py

```diff
@@ -1421,3 +1564,153 @@ def _vander_default(x, N=None, increasing=False):
 @all_of_type(ndarray)
 def vander(x, N=None, increasing=False):
     return (x,)
+
+
+@create_numpy(_first2argreplacer)
+@all_of_type(ndarray)
+def split(ary, indices_or_sections, axis=0):
+    return (ary, indices_or_sections)
+
+
+@create_numpy(_first2argreplacer)
+@all_of_type(ndarray)
+def array_split(ary, indices_or_sections, axis=0):
+    return (ary, indices_or_sections)
+
+
+@create_numpy(
+    _first2argreplacer,
+    default=lambda ary, indices_or_sections: split(ary, indices_or_sections, axis=2),
+)
+@all_of_type(ndarray)
+def dsplit(ary, indices_or_sections):
+    return (ary, indices_or_sections)
+
+
+@create_numpy(
+    _first2argreplacer,
+    default=lambda ary, indices_or_sections: split(ary, indices_or_sections, axis=1),
+)
+@all_of_type(ndarray)
+def hsplit(ary, indices_or_sections):
+    return (ary, indices_or_sections)
+
+
+@create_numpy(
+    _first2argreplacer,
+    default=lambda ary, indices_or_sections: split(ary, indices_or_sections, axis=0),
+)
+@all_of_type(ndarray)
+def vsplit(ary, indices_or_sections):
+    return (ary, indices_or_sections)
+
+
+def _tile_default(A, reps):
+    if isinstance(reps, numbers.Integral):
+        reps = (reps,)
+
+    prep = len(reps) - ndim(A)
+    if prep > 0:
+        for _ in range(prep):
+            A = expand_dims(A, 0)
+    elif prep < 0:
+        reps = tuple(itertools.repeat(1, abs(prep))) + reps
+
+    for axis, dim in enumerate(reps):
+        dim = operator.index(dim)
+        if dim == 0:
+            return asarray([], dtype=A.dtype)
+
+        tile = A
+        for _ in range(dim - 1):
+            A = concatenate((A, tile), axis=axis)
+
+    return A
+
+
+@create_numpy(_first2argreplacer, default=_tile_default)
+@all_of_type(ndarray)
+def tile(A, reps):
+    return (A, reps)
+
+
+@create_numpy(_first2argreplacer)
```

Comment:
`repeat` can have a fairly simple default in terms of `concatenate`.

Reply:
`np.concatenate((a,) * n_repeats)` for the 1D repeat case.

Reply:
I think `concatenate` can't be used here as `repeat` repeats specified elements adjacently. Here is an example:

```python
In [2]: a = onp.array([1, 2, 3])                                                

In [3]: onp.concatenate((a,) * 3)                                               
Out[3]: array([1, 2, 3, 1, 2, 3, 1, 2, 3])

In [4]: onp.repeat(a, 3)                                                        
Out[4]: array([1, 1, 1, 2, 2, 2, 3, 3, 3])
```

My original idea was to use `insert` and `itertools.repeat` but not sure about this either.

Reply:
You could do this with slicing:
```python
new_len = len(a) * repeat
res = np.empty_like(a, shape=new_len)
for i in range(repeat):
    res[i::repeat] = a
```

It would take more code to generalise to n-dimensions, but should be do-able.

Reply:
I've pushed the default for `repeat`. I think the case where the `repeats` arg is an array of ints would involve iterating over array elements so I didn't include it in the implementation.

Reply:
I would just remove it, since it's still a loop.

Reply:
Remove `_repeats_default` entirely?

Reply:
Correct. :c